File: app/src/main/java/com/estrrado/tradopay/ui/modules/home/adapter/SearchSampleItemAdapter.java
package com.estrrado.tradopay.ui.modules.home.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.estrrado.tradopay.ui.modules.OnItemSelectedlistner;
import com.example.tradopay.R;

public class SearchSampleItemAdapter extends RecyclerView.Adapter<SearchSampleItemAdapter.SearchSampleViewHolder> {
    OnItemSelectedlistner onItemSelectedlistner;
    String[] arr;
    public SearchSampleItemAdapter(OnItemSelectedlistner onItemSelectedlistner,String[] arr) {
        this.onItemSelectedlistner=onItemSelectedlistner;
        this.arr= arr;

    }

    @NonNull
    @Override
    public SearchSampleViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new SearchSampleViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.search_sample_item,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull SearchSampleViewHolder holder, int position) {

        holder.txtSearchTrend.setText(arr[position]);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemSelectedlistner.onItemClick(null,position);
            }
        });

    }

    @Override
    public int getItemCount() {
        return arr.length;
    }

    public class SearchSampleViewHolder extends RecyclerView.ViewHolder
    {
        TextView txtSearchTrend;
        public SearchSampleViewHolder(@NonNull View itemView) {
            super(itemView);
            txtSearchTrend=itemView.findViewById(R.id.txtSearchTrend);
        }
    }
}

File: app/src/main/java/com/estrrado/tradopay/adapter/SearchSampleItemAdapter.kt
package com.estrrado.tradopay.ui.modules.home.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.estrrado.tradopay.listener.OnItemSelectedlistner
import com.estrrado.tradopay.ui.modules.home.adapter.SearchSampleItemAdapter.SearchSampleViewHolder
import com.example.tradopay.R

class SearchSampleItemAdapter(var onItemSelectedlistner: OnItemSelectedlistner, var arr: Array<String>) : RecyclerView.Adapter<SearchSampleViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchSampleViewHolder {
        return SearchSampleViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.search_sample_item, parent, false))
    }

    override fun onBindViewHolder(holder: SearchSampleViewHolder, position: Int) {
        holder.txtSearchTrend.text = arr[position]
        holder.itemView.setOnClickListener { onItemSelectedlistner.onItemClick(null, position) }
    }

    override fun getItemCount(): Int {
        return arr.size
    }

    inner class SearchSampleViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var txtSearchTrend: TextView

        init {
            txtSearchTrend = itemView.findViewById(R.id.txtSearchTrend)
        }
    }
}